### ISSUE TITLE
Streaming API does not seem to be thread-safe

### ISSUE THREAD

**Issue by tillahoffmann** [mozilla/DeepSpeech#2328]
Using the streaming api in `deepspeech==0.5.0` does not seem to be thread safe. It seems that [`https://github.com/mozilla/DeepSpeech/blob/master/native_client/test/concurrent_streams.py`](https://github.com/mozilla/DeepSpeech/blob/master/native_client/test/concurrent_streams.py) tries to address a similar question, but I couldn't figure out where my code differs substantially from the linked code. See below for a (hopefully) reproducible example.

### Setup code to load the model

```python
import wave
import deepspeech as ds
import numpy as np

# Beam width used in the CTC decoder when building candidate transcriptions
BEAM_WIDTH = 500

# The alpha hyperparameter of the CTC decoder. Language Model weight
LM_ALPHA = 0.75

# The beta hyperparameter of the CTC decoder. Word insertion bonus.
LM_BETA = 1.85

# These constants are tied to the shape of the graph used (changing them changes
# the geometry of the first layer), so make sure you use the same constants that
# were used during training

# Number of MFCC features to use
N_FEATURES = 26

# Size of the context window used for producing timesteps in the input vector
N_CONTEXT = 9

SAMPLE_RATE = 16000
BLOCK_SIZE=10

model_path = "/models/deepspeech-0.5.0-models/output_graph.pbmm"
alphabet_path = "/models/deepspeech-0.5.0-models/alphabet.txt"
language_model_path = "/models/deepspeech-0.5.0-models/lm.binary"
trie_path = "/models/deepspeech-0.5.0-models/trie"

model = ds.Model(model_path, N_FEATURES, N_CONTEXT, alphabet_path, BEAM_WIDTH)
model.enableDecoderWithLM(alphabet_path, language_model_path, trie_path, LM_ALPHA, LM_BETA)

with wave.open('can-you-keep-a-secret.wav') as audio:
    frames = audio.readframes(audio.getnframes())
    samples = np.frombuffer(frames, dtype=np.int16)
```

### Sequential processing with desired output

```python
for _ in range(4):
    context = model.setupStream()
    model.feedAudioContent(context, samples)
    print(model.finishStream(context))
```

```
can you keep a secret
can you keep a secret
can you keep a secret
can you keep a secret
```

### Simulated parallel processing with unexpected output

``` python
contexts = [model.setupStream() for _ in range(4)]
batches = np.array_split(samples, 10)

for batch in batches:
    for context in contexts:
        model.feedAudioContent(context, batch)
        
for context in contexts:
    print(model.finishStream(context))
```

```
can you keep a secret
you can sleep arete g
you can separate 
you can sleep ready 
```

### Background information

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: written custom code to use DeepSpeech, but did not modify DeepSpeech
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04 in docker container with Mac OSX 10.14.5 host and docker desktop community edition 2.1.0.1 (37199)
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: not installed
- **TensorFlow version (use command below)**: NA
- **Python version**: 3.6.7
- **Bazel version (if compiling from source)**: not compiled from source
- **GCC/Compiler version (if compiling from source)**: not compiled from source
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: NA
- **Exact command to reproduce**: see above

**Comment by kdavis-mozilla** [comment]
This is in the [Known Issues section of the 0.5.1 release notes](https://github.com/mozilla/DeepSpeech/releases/tag/v0.5.1).

**Comment by tillahoffmann** [comment]
Great, thanks for the pointer, @kdavis-mozilla. Should I close this issue in favour of an existing one?

**Comment by reuben** [comment]
Yes, it's already been fixed in master and should go out with 0.6.0. In the mean time, you can use one of the 0.6 alphas with the re-exported 0.5.1 model: https://github.com/reuben/DeepSpeech/releases/v0.6.0-alpha.4

**Comment by reuben** [comment]
Fix was in #2146 FWIW.
